src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -326,6 +326,49 @@ public boolean isTableCompatible(final String tableName) {
     return compatible;
   }
 
+  @Override
+  public void appendMetaColIfNotExist(final String tableName) {
+    checkConnection();
+    InternalUtils.assertNotEmpty("tableName", tableName);
+    String query = "desc table identifier(?)";
+    PreparedStatement stmt = null;
+    ResultSet result = null;
+    boolean hasMeta = false;
+    boolean isVariant = false;
+    try {
+      stmt = conn.prepareStatement(query);
+      stmt.setString(1, tableName);
+      result = stmt.executeQuery();
+      while (result.next()) {
+        if (result.getString(1).equals("RECORD_METADATA")) {
```

Comment:
consider extracting "RECORD_METADATA" to a static final string, something like
```
private static final String METADATA_COLUMN = "RECORD_METADATA"
```

and then using it here. I know table creation has not used it but may be we can start.